src/core.py

```diff
@@ -168,11 +168,13 @@ def recent_blobs(self):
 
     def _listen(self, last_object, object_type, blocking=True, interval=15, query=None):
         if last_object is None:
-            last_object = next(self._recent(object_type, query=query))
+            last_object = next(self._recent(object_type, query=query), None)
             # If there are no elements (even first element): just get new samples from now on
             if last_object is not None:
                 last_id = last_object.id
                 last_time = last_object.upload_time
+            else:
+                last_id = last_time = None
```

Comment:
maybe don't initialise them in `else`, and instead initialise them at the beginning of the method? It'll save some brainpower programmers ensuring that they're always initialised.

Reply:
or maybe even better, move all of this to a separate private method `_get_last_time()` or similar - but whichever you prefer

Reply:
+1 for initializing them to `None` at the beginning, a nested if that assigns the value seems quite complicated

Reply:
Moving all this logic to `_get_last_object(last_object: Union[str, MWDBObject]) -> MWDBObject` also work